src/scripts/object.search.js

```diff
@@ -90,15 +90,19 @@ module.exports = (robot, res) => {
 						message: classifierResult.description
 					});
 
-					return [];
+					return matches;
 				}
 				else if (classifierResult.classify_result.classes) {
 					let count = 0;
 					context.robot.logger.debug(`${TAG}: Classify results for searchPhrase ${searchPhrase}`);
 					_.forEach(classifierResult.classify_result.classes, (classifier) => {
+						let previous = (count === 0) ? undefined : classifierResult.classify_result.classes[count - 1];
+						let confidenceDifference = (!previous) ? 0 : previous.confidence - classifier.confidence;
```

Comment:
You have no checks for these fields:
 -previous.confidence
- classifier.confidence